dns/dyndns/src/etc/inc/plugins.inc.d/dyndns/phpDynDNS.inc

```diff
@@ -387,7 +399,7 @@ class updatedns
             log_error("Dynamic DNS ({$this->_dnsHost} via {$this->_dnsServiceList[$this->_dnsService]}): _update() starting.");
         }
 
-        if ($this->_dnsService != 'ods' and $this->_dnsService != 'route53' and $this->_dnsService != 'route53-v6') {
+        if ($this->_dnsService != 'ods' and $this->_dnsService != 'netcup' and $this->_dnsService != 'netcup-v6' and $this->_dnsService != 'route53' and $this->_dnsService != 'route53-v6') {
```

Comment:
I am afraid this is getting a bit out of hand now with hand-rolling the curl request inline as the purpose of this library was to reduce code weight. In case of route53 there's an extra class to handle this and we should figure out a way to integrate new services a bit more gracefully even if we have to add new interfaces for it so we don't have to touch a lot of "if-else" constructs along the way.

Reply:
Maybe it are just as much if-else clauses because there are so many providers supported :wink:
- Well the ability to extend with new dyndns modules is limited in current framework as many REST APIs need multiple calls for authentication, getting information and submitting changes.
- The existing structure does either allow only one curl request or one final error state to be reported back. I could rework the code to move the `curl_open` and `curl_close` in its own if-clause like `if ($this->_usecurlhandler)` but then every module would have to be touched. Then at least, every plugin provider could decide on its own and flip bits as with `$this->_useIPv6`.
- Maybe it would be best to create a new `updatedns` sub-class `dyndnsprovider` stanza thingy which handles the updating internally and is interacted upon with `construct/init()`, `run()`, `getstate()`, `geterror()` etc.
It would be quite an effort as all modules woud have to be re-wrapped, and I can think of doing it but would need more advice/documentation on current class model. Even If i migrate all code in an extra class like r53 it will still need the if-clause here to exempt the curl handling. 

**But, it would be really nice to get things rolling with netcup support. Can this PR be merged once all other issues are fixed?**